classes/core/traits/HasParent.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * @file classes/core/traits/HasParent.php
+ *
+ * Copyright (c) 2022 Simon Fraser University
+ * Copyright (c) 2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class HasParent
+ * @ingroup core_traits
+ *
+ * @brief HasParent trait.
+ *
+ */
+
+namespace PKP\core\traits;
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\core\DataObject;
+
+trait HasParent
```

Comment:
Sorry to change things on you last minute, but since this will live in `PKP\core`, maybe we should clarify what it's meant to be used on? Perhaps we could call it `EntityHasParent` or `EntityWithParent`?

Reply:
I will then call it EntityWithParent...